### ISSUE TITLE
Reading a sequence of json value from a bytes reader

### ISSUE THREAD

**Issue by mbarbin** [dbuenzli/jsont#4]
I am looking for a helper that is the jsont conceptual equivalent of something like this from [atd](https://github.com/ahrefs/atd/blob/6a55687736518c8e505c9f882a67f704ec335048/atdgen-runtime/src/util.mli#L127):

```ocaml
  val seq_from_lexbuf :
    ?fin:(unit -> unit) ->
    'a reader -> Yojson.Safe.lexer_state -> Lexing.lexbuf -> 'a Seq.t
```

I suppose you can call `Jsont_bytesrw.decode` in a loop until you reach `Reader.eod` but I am not exactly sure how to best implement the lookahead part to check for eod (and Jsont errors are opaque).

Would you be able to suggest a way this can be done with the current API? (this would help me to continue my exploration of the jsont library, any hacks welcome).

Maybe that's an idea for some convenience helpers down the road (something to consider).

Thank you!

**Comment by dbuenzli** [comment]
From the README:

> Questions are welcome but better asked on the [OCaml forum](https://discuss.ocaml.org/) than on the issue tracker.

Please next time ask them there.

I don't know what syntax this is parsing. But there is no such thing in JSON. `Jsont_bytesrw` being standard compliant when you reach the end of stream you need to have parsed a JSON value according to the specification.

The nearest you could have is [JSON text sequences](https://www.rfc-editor.org/rfc/rfc7464) (very rarely used as far as I know) or [JSON lines](https://jsonlines.org/). In which case you should likely have a reader that chunks into sub readers based on the unambiguous delimiters (it is in the case of JSON lines because the whole value has to be on a single line).

Basically a combinator: 

```
val split_reader : on_char:Char.t -> Bytes.Reader.t -> Bytes.Reader.t Seq.t 
```

You can then invoke `Jsont_bytesrw.decode` on the subreaders.

**Comment by dbuenzli** [comment]
Also if this is an open JSON array another thing you could try is to define you own toplevel [array map](https://erratique.ch/software/jsont/doc/Jsont/Array/index.html) which side effects or inverts control with effects on decode elements and lazily computes elements or inverts control on encodes.

You are then back to a single `Jsont_bytesrw.decode` invocation and emit standard JSON.

**Comment by mbarbin** [comment]
> Please next time ask them there.

Will do.

> you could try is to define you own toplevel [array map](https://erratique.ch/software/jsont/doc/Jsont/Array/index.html)

Yes this is something like an open JSON array, although in this instance I don't think it necessarily has to be. I did not realize I was out of common practice or spec (for example `jq . my-file` works on such sequenced file even when each expr is pretty-printed and spans multiple lines). That's useful input for me to think about, and I might end up changing things around. Thank you!
